### DISCUSSION TITLE
Get Software GUID on inventory  [Q&A]

### DISCUSSION THREAD

**Discussion by EmilianoCagliari** [glpi-project/glpi-agent#726] [Q&A] [answered]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

Hi, i dont know if this could be a feature. But in my case i deploy the software package and install with msiexec and /quiet parameter but if i have to uninstall i need the software guid to execute the msiexec /x {GUID}  to uninstall.
And i dont know where this info is in GLPI's Software tab ( If this data is collected) but if this is not collected it could be possible? Or there is another way to do this?  

Thanks in advance!

Regards!

**Comment by g-bougard** [comment, marked as answer]
Hi @EmilianoCagliari 

GUID is collected by GLPI-Agent and it is visible in software inventory only. I checked GLPI code and it only uses it to set software name if software has no name. So you have no way to get it automatically in GLPI.

Anyway, I think you can get it directly from the package you installed if you're only interested by uninstalling software you installed with Deploy task. You should have some windows tools to extract this data from the package, but I don't know them myself.
If you're also interested to uninstall software still installed, you'll have to get a full inventory of the computer and check the software category.
Eventually, you can run a software partial inventory and extract the data from the generated json. I imagine you can even do that during a Deploy task but I'm sure this will slow down your process.
Finally you can use some logic in your Deploy task to extract the GUID as agent just do, check this out here: https://github.com/glpi-project/glpi-agent/blob/de5f5b515e2352c3fa864f5cbe9cd8f735ae6fff/lib/GLPI/Agent/Task/Inventory/Win32/Softwares.pm#L195
To summarize, agent takes keys from `HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/Windows/CurrentVersion/Uninstall` as GUID, and report as NAME the "DisplayName" field.

Maybe [Collect task](https://faq.teclib.com/03_knowledgebase/inventory/Collect_data/) can help you to list all GUID with included DeisplayName.

**Reply by EmilianoCagliari** [reply]
Hi @g-bougard !  

I will try to make some logic to get the GUID with your information. Im waiting my boss confirmation to do that but anyway i will try by my own.

Thank you for your response.
